doc/Factory-Compatibility.md

```diff
@@ -0,0 +1,768 @@
+| ---------- | -------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| geos       | `RGeo::Geos.factory`                                                                         | 2D Cartesian factory backed by the GEOS C API                                                                                             |
+| geos_zm    | `RGeo::Geos.factory(has_z_coordinate: true, has_m_coordinate: true)`                         | 4D Cartesian factory backed by the GEOS C API                                                                                             |
+| ffi        | `RGeo::Geos.factory(has_z_coordinate: true, has_m_coordinate: true)`                         | 2D Cartesian factory backed by the GEOS C API (via FFI)                                                                                   |
+| ffi_zm     | `RGeo::Geos.factory(has_z_coordinate: true, has_m_coordinate: true, native_interface: :ffi)` | 4D Cartesian factory backed by the GEOS C API (vi FFI)                                                                                    |
+| cartesian  | `RGeo::Cartesian.simple_factory`                                                             | 2D Cartesian factory implemented in Ruby                                                                                                  |
+| spherical  | `RGeo::Geographic.spherical_factory`                                                         | Factory with calculations performed on a spherical surface, written in Ruby                                                               |
+| projection | `RGeo::Geographic.simple_mercator_factory`                                                   | Compound factory that uses a spherical factory to create geometries, but a geos or cartesian factory for computations on a projected coordinate system |
+
+For more information about choosing a factory, see the [Choosing a Fatory doc](https://github.com/rgeo/rgeo/wiki/Which-factory-should-I-use%3F-(draft)).
+
+## Tested Methods
+
+The table tests all of the methods defined in the [OGC SFA v1.1](https://www.ogc.org/standards/sfa) for each geometry type. In the case of methods that require another geometry (ex. `LineString#interects?(Polygon)`), all geometry types will be tested as the parameter.
+
+_Note: This list is not exhaustive of all the methods defined by each factory. The geos factory in particular has a good amount of non-standard methods implemented. These can be found in the YARD docs._
+<!-- TODO: this is not true, C methods are not listed in documentation, we should do something about it. -->
```

Comment:
@keithdoggett @JoakimKlaxit FYI. We should dig into that

Reply:
Yes, agreed. I've gotten this to work on my package before, so it may just be touching the format up. [code](https://github.com/keithdoggett/spatial_stats/blob/master/ext/spatial_stats/csr_matrix.c#L112) : [corresponding docs](https://keithdoggett.github.io/spatial_stats/SpatialStats/Weights/CSRMatrix.html#initialize-instance_method)

Reply:
@keithdoggett nice ! I'd love to know the solution for my own projects !